src/vsr/replica_format.zig

```diff
@@ -27,205 +23,288 @@ pub fn format(
 
     var replica_format = ReplicaFormat{};
 
-    try replica_format.format_wal(allocator, options.cluster, storage);
-    assert(!replica_format.formatting);
-
-    try replica_format.format_replies(allocator, storage);
-    assert(!replica_format.formatting);
-
-    try replica_format.format_grid_padding(allocator, storage);
-    assert(!replica_format.formatting);
+    var arena = std.heap.ArenaAllocator.init(allocator);
+    defer arena.deinit();
 
-    superblock.format(
-        ReplicaFormat.format_superblock_callback,
-        &replica_format.superblock_context,
-        options,
-    );
-
-    replica_format.formatting = true;
-    while (replica_format.formatting) storage.run();
+    try replica_format.queue_format_wal(arena.allocator(), options.cluster, storage);
+    replica_format.format_and_tick(storage, &superblock, options);
+    replica_format.verify_writes();
 }
 
+/// When formatting, we write:
+/// * constants.journal_slot_count many prepares,
+/// * 1 write that contains all of the headers,
+pub const writes_max = constants.journal_slot_count + 1;
+
 fn ReplicaFormatType(comptime Storage: type) type {
     const SuperBlock = vsr.SuperBlockType(Storage);
     return struct {
         const ReplicaFormat = @This();
+        const SectorBitSet = std.bit_set.ArrayBitSet(
+            usize,
+            @divExact(data_file_size_min, constants.sector_size),
+        );
```

Comment:
Let's use dynamic bitset here, we don't gain anything by adding another copy of bitset's code to our binary.